### DISCUSSION TITLE
Using rye to manage global Python installtions like nvm

### DISCUSSION THREAD

**Discussion by itay-raveh** [astral-sh/rye#70] [Q&A]
When I used `rye install` it linked the tool `pyvenv.cfg` to a rye python installation, but obviously `$ which python3` still gives `/usr/bin/python3`.

I could manually use an alias or similar, but the question is if rye can be used like [nvm](https://github.com/nvm-sh/nvm), which has `nvm use <version>`.

Should rye be able to do this, or is the scope limited to specific projects?

**Comment by mitsuhiko** [comment]
If you add `rye` to the `PATH` it should work:

```
export PATH="$HOME/.rye/shims:$PATH"
```

Afterwards `python` and `python3` will pick up the local `.python-version` defined in a folder, or if none is defined will fall back to a global python installation that's on the path to not disrupt the already existing system.

**Reply by itay-raveh** [reply]
This is what I already have. But I don't have a `.python-version` in every folder like I don't have `.nvmrc` or `.node-version` in every folder. 

I want to be able to set a rye Python version as the fallback for when no `.python-version` is found, instead of the system version, the same way I would with `nvm use`.

**Reply by mitsuhiko** [reply]
It walks all the way up. While not perfect, you should be able to place a `.python-version` in the root folder or your home directory to configure this. But I can see the utility of ensuring that a rye provided python can be used anywhere.

Let me think about this.

**Reply by skippednote** [reply]
@mitsuhiko I have generated a `.python-version` in my home directory using `rye pin 3.12` and have installed the Python version using `rye fetch 3.12.0`. However, when I run `python -V` I get `error: target shim binary not found` and `python3` points to system python. What am I missing? I have the shim's directory in my PATH already.
